src/sphinx_autodoc_typehints/__init__.py

```diff
@@ -556,13 +556,22 @@ def _inject_types_to_docstring(
                 insert_index = None
                 break
             elif line.startswith(":return:") or line.startswith(":returns:"):
+                # This is brittle, I have not found information on how standard the use of " -- " to
+                # separate type from description is.
+                if " -- " in line and not app.config.typehints_use_rtype:
```

Comment:
I'd like something better than this 🤔

Reply:
Remove the comment and we're good.

Reply:
I also added a test for this case (using `Union[int, str]` as an example).